test/test_ops.py

```diff
@@ -194,7 +195,11 @@ def to_cpu(arg):
             self.assertEqual(len(cuda_results), len(cpu_results))
 
             for cpu_result, cuda_result in zip(cpu_results, cuda_results):
-                self.assertEqual(cpu_result, cuda_result)
+                if not op.deterministic and isinstance(cpu_result, torch.Tensor):
```

Comment:
If the op is non deterministic and the result is not a tensor the else condition will catch. We probably don't want to do equality comparison on the results of non-deterministic ops, I can't think of any examples off the top of my head that would trigger this but better to be precise.

Reply:
were you thinking of something like:

```py
                if op.deterministic:
                    self.assertEqual(cpu_result, cuda_result)
                elif isinstance(cpu_result, torch.Tensor):
                    self.assertEqual(cpu_result.dtype, cuda_result.dtype)
                    self.assertEqual(cpu_result.shape, cuda_result.shape)
                else:
                    # This is not a current use-case, all non-deterministic ops return a tensor
                    pass
```